app/models/post.rb

```diff
@@ -1,4 +1,15 @@
 class Post < ApplicationRecord
-  has_many :comments, :likes
-  belongs_to :user, counter_cache: true
+  has_many :comments
+  has_many :likes
+  belongs_to :user, foreign_key: 'user_id', counter_cache: true, optional: true
+
+  after_save :update_post_counter
```

Comment:
- [ ] Kindy check on your method spelling its different from the update_counter method you created.